src/OpenTelemetry/BaseProcessor.cs

```diff
@@ -41,7 +43,7 @@ public BaseProcessor()
 #else
     internal
 #endif
-        int Weight { get; set; }
+        ProcessorPipelineWeight PipelineWeight { get; set; } = ProcessorPipelineWeight.PipelineMiddle;
```

Comment:
I think I understand the intention of introducing this concept. I don't feel I would agree/approve it (I have trouble with this direction), and considering time constraints, consider removing it from this PR.

Reply:
There is no way to force something to the back of the pipeline currently. If I remove this, this PR and all of its scenarios need to be reworked. Here are some options:

1) We can make `PipelineWeight` purely internal in SDK and always expose SDK internals to OtlpExporter. No issue with this, just that components (like `AutoFlushActivityProcessor`) can never be made to work together without a way to participate in the ordering. The reason I felt exposing it experimentally was a good idea is we can see if it works and then bring learnings back to the spec. But we could always tackle that whole effort separately.

2) We can come up with some other way to do it. Likely will also need an API though so we could end up back in the same spot.

3) Rework this PR and its scenarios so it doesn't try to add itself to end of the pipeline.

Quickest option would probably be 1.

Reply:
I'm fine as long as there is no public API (even experimental one) for "weight".

Reply:
Updated: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5400/commits/c2db1009e350045b1765456c60aee948c3d572d9